source/components/atoms/Select/Select.tsx

```diff
@@ -43,59 +43,74 @@ interface Props {
   style?: ViewStyle;
   onBlur: (value: string) => void;
   onOpen: (event: undefined, isSelect: true) => void;
+  onClose: (event: undefined, isSelect: true) => void;
   showErrorMessage?: boolean;
   error?: { isValid: boolean; message: string };
 }
 
-const Select: React.FC<Props> = React.forwardRef(({
-  items,
-  onValueChange,
-  onBlur,
-  onOpen,
-  placeholder,
-  value,
-  editable = true,
-  showErrorMessage = true,
-  error,
-  style,
-}, ref) => {
+const Select: React.FC<Props> = React.forwardRef(
+  (
+    {
+      items,
+      onValueChange,
+      onBlur,
+      onOpen,
+      onClose,
+      placeholder,
+      value,
+      editable = true,
+      showErrorMessage = true,
+      error,
+      style,
+    },
+    ref
+  ) => {
+    const currentItem = items.find((item) => item.value === value);
+    const handleValueChange = (itemValue: string | number | boolean) => {
+      if (onValueChange && typeof onValueChange === "function") {
```

Comment:
it feels strange to check if the type of onValueChange is a function when its typed as a function in typescript. Or do you want to be more safe during runtime?